### PR TITLE
PERF: Create a partial regular post_search_data index on large sites.

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

File: app/jobs/scheduled/create_recent_post_search_indexes.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      SiteSetting.search_recent_regular_posts_offset_post_id = offset_post_id(private_message: false)
+
+      DB.exec(<<~SQL)
+      DROP INDEX IF EXISTS temp_idx_recent_regular_post_search_data;
+      SQL
+
+      DB.exec(<<~SQL)
+      CREATE INDEX #{Rails.env.test? ? '' : 'CONCURRENTLY'} temp_idx_recent_regular_post_search_data
+      ON post_search_data USING GIN(search_data)
+      WHERE NOT private_message AND post_id >= #{SiteSetting.search_recent_regular_posts_offset_post_id}
```
Comment (eviltrout) [inline app/jobs/scheduled/create_recent_post_search_indexes.rb:29]: I recommend always calling `to_i` with SQL string interpolation of numbers. If someone could change that site setting to a string they could inject SQL.

File: app/jobs/scheduled/create_recent_post_search_indexes.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+      DB.exec(<<~SQL)
+      #{Rails.env.test? ? '' : "BEGIN;"}
+      DROP INDEX IF EXISTS #{REGULAR_POST_SEARCH_DATA_INDEX_NAME};
+      ALTER INDEX temp_idx_recent_regular_post_search_data RENAME TO #{REGULAR_POST_SEARCH_DATA_INDEX_NAME};
+      #{Rails.env.test? ? '' : "COMMIT;"}
+      SQL
+    end
+
+    def offset_post_id(private_message:)
```
Comment (eviltrout) [inline app/jobs/scheduled/create_recent_post_search_indexes.rb:40]: This function is only ever called once with `false` as the value so why make it a parameter?

File: app/jobs/scheduled/create_recent_post_search_indexes.rb (hunk 1 above)
Comment (SamSaffron) [reply app/jobs/scheduled/create_recent_post_search_indexes.rb:29]: In this case you are doing DB.exec ... so `DB.exec(<<~SQL, post_id: SiteSetting.....)` will do the trick

File: app/jobs/scheduled/create_recent_post_search_indexes.rb (hunk 2 above)
Comment (SamSaffron) [reply app/jobs/scheduled/create_recent_post_search_indexes.rb:40]: agree... lest just call it `regular_offset_post_id`

File: app/jobs/scheduled/create_recent_post_search_indexes.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      end
+
+      SiteSetting.search_recent_regular_posts_offset_post_id = offset_post_id(private_message: false)
+
+      DB.exec(<<~SQL)
+      DROP INDEX IF EXISTS temp_idx_recent_regular_post_search_data;
+      SQL
+
+      DB.exec(<<~SQL)
+      CREATE INDEX #{Rails.env.test? ? '' : 'CONCURRENTLY'} temp_idx_recent_regular_post_search_data
```
Comment (SamSaffron) [inline app/jobs/scheduled/create_recent_post_search_indexes.rb:27]: let's not call the index "temp" it is strictly temporary ... but it is always there... people may thing they created it by accident..

`idx_recent_regular_post_search_data` is a good name

File: app/jobs/scheduled/create_recent_post_search_indexes.rb (hunk 1 above)
Comment (eviltrout) [reply app/jobs/scheduled/create_recent_post_search_indexes.rb:29]: Even better!

**Comment by tgxworld** [comment]
@eviltrout @SamSaffron Thank you for reviewing :+1:

**Pull request merged**